packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -1130,17 +1154,62 @@ class DoNothingIntent extends Intent {
   const DoNothingIntent._();
 }
 
-/// An [Action], that, as the name implies, does nothing.
+/// An [Intent], that, as the name implies, is bound to a [DoNothingAction],
+/// but, in addition to not performing an action, also stops the propagation of
+/// the key event bound to this intent to other key event handlers in the focus
+/// chain.
 ///
-/// Attaching a [DoNothingAction] to an [Actions] mapping is one way to disable
-/// an action defined by a widget higher in the widget hierarchy.
+/// Attaching a [DoNothingAndStopPropagationIntent] to a [Shortcuts.shortcuts]
+/// mapping is one way to disable a keyboard shortcut defined by a widget higher
+/// in the widget hierarchy. In addition, the bound [DoNothingAction] will
+/// return false from [DoNothingAction.shouldHandleKey], causing the key bound to
+/// this intent to be passed on to the platform embedding as "not handled" with
+/// out passing it to other key handlers in the focus chain (e.g. parent
+/// `Shortcuts` widgets higher up in the chain).
 ///
-/// This action can be bound to any intent.
+/// This intent cannot be subclassed.
 ///
 /// See also:
-///  - [DoNothingIntent], which is an intent that can be bound to a keystroke in
+///
+///  * [DoNothingIntent], a similar intent that will handle the key event.
+class DoNothingAndStopPropagationIntent extends Intent {
+  /// Creates a const [DoNothingAndStopPropagationIntent].
+  factory DoNothingAndStopPropagationIntent() => const DoNothingAndStopPropagationIntent._();
+
+  // Make DoNothingAndStopPropagationIntent constructor private so it can't be subclassed.
+  const DoNothingAndStopPropagationIntent._();
+}
+
+/// An [Action], that, as the name implies, doesn't do anything when invoked.
+///
+/// Attaching a [DoNothingAction] to an [Actions.actions] mapping is a way to
+/// disable an action defined by a widget higher in the widget hierarchy.
+///
+/// If [shouldHandleKey] returns false, then not only will this action do
+/// nothing, but it will also terminate the propagation of the key event used to
```

Comment:
"also terminate" -> isn't it always terminating the propagation when an action is found regardless of this flag? The flag just controls whether we tell the embedding if the event was handled or not?

Reply:
Good point.  Updated the docs.